### PR TITLE
Weather App Features

### PR DESCRIPTION
**Pull request by george-shammar**
This Pull Request seeks to add the following features to the default branch:
1. Functionalities that retrieves weather date from openWeather API.
2. Functionalities that displays the retrieved data and appropriately styled.
3. Button that converts temperature from Celcius to Fahrenheit.

### CODE REVIEW

**Review by trekab — CHANGES_REQUESTED** [review]
### Changes Required! :recycle: 
 Hi @george-shammar! Well done with the UI! It looks nice. You did well working with the asynchronous code. However, there are some changes I'd like you to work on before having your project approved. These include:
- [ ] Delete the `node_modules` folder from your repository.
- [x] Implement a toggle control to determine if the result will be shown in Celsius or Fahrenheit.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app)). Your current implementation is just a  button that converts to Fahrenheit without a possibility of converting back to Celsius.
- [x] Your app should change its appearance based on the received data ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
- [x] Try to look into your error handling. When I submit wrong/non-existing cities, I get a type error in the console. Your app would be more user friendly if at least the user got some helpful message from the UI.([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/error_handling_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622133/))
- [ ] __[Optional]__ Add a ‘loading’ component that displays from the time the form is submitted until the information comes back from the API ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/loading_component_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
- [ ] __[Optional]__ Fetch the current geolocation of the user when the app is first accessed and fetch the weather from that location.

__Please make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__[Treasure Kabareebe](https://trekab.github.io)__

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @george-shammar, congratulations on this project :1st_place_medal: The node_modules have been removed as pointed out earlier, the toggle functionality fully implemented :+1: Errors properly handled with a  beautiful UI text informing the user about the said error :smile: However, there is one functionality preventing us from approving the project :slightly_smiling_face: Kindly check the comment below:

- There should be a change in background appearance once data is retrieved. This functionality has not been addressed, let us kindly fix it before submitting another code review request

[Optional]
- Kindly use the right linter configuration. Find it [here](https://github.com/microverseinc/linters-config/blob/master/javascript/.github/workflows/linters.yml)


Once this is addressed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "parserOptions": {
+      "ecmaVersion": 2018,
+      "sourceType": "module"
+    },
+    "extends": ["airbnb-base"],
+    "rules": {
+      "no-shadow": "off",
+      "no-param-reassign": "off",
+      "eol-last": "off",
+      "arrow-parens": "off"
```
Comment (addod19) [inline .eslintrc.json:17]: Do not make any changes to the linter files. FInd it [here](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json)

**Comment by george-shammar** [comment]
@addod19 , thanks for the review. The background is changing based received data as required. Perhaps, you're supplying names of cities with similar weather data conditions. Try cities with wide weather ranges e.g Warsaw, Mecca, Jeddah, Auckland, Munich. The backgroundChange function is based on main temperature ranges.

I have also changed to the latest linters config. I used the outdated one initially.

Many thanks. :smile:

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏
# OPTIONAL
- [ ] It's highly recommended that you include all the steps to run the project locally in README. Your app can not be installed without running `npm install`.
### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Pull request merged**